### PR TITLE
feat(ui): port quality-profile management from legacy UI

### PR DESCRIPTION
**Pull request by bassings**
## What & Why

Ports the **first critical gap** from the UI-MIGRATION backlog (`specs/UI-MIGRATION.md`): quality-profile management. The modern UI had no way to create, edit, reorder, or delete quality profiles — users were stuck with whatever defaults existed at startup.

## API Contract Discovered

From `couchpotato/core/plugins/profile/main.py` and `quality/main.py`:

| Endpoint | Method | Key fields |
|---|---|---|
| `profile.list` | GET | `list[].{_id, label, order, core, hide, qualities[], finish[], 3d[], wait_for[], stop_after[]}` |
| `quality.list` | GET | `list[].{identifier, label, hd, allow_3d, size_min, size_max, order}` |
| `profile.save` | POST | `id?, label, minimum_score, wait_for, stop_after, types (JSON array of {quality, finish, 3d})` |
| `profile.save_order` | POST | `ids[], hidden[]` (parallel arrays) |
| `profile.delete` | POST | `id` |

`wait_for` / `stop_after` are single top-level values applied to all qualities. The first quality always gets `finish=true` (backend enforces, UI mirrors).

## Files Changed

- `specs/UI-PORT-01-quality-profiles.md` — full spec with API contract
- `couchpotato/static/scripts/ui/profile-editor.js` — pure ES module: `profileToForm`, `formToPayload`, `addQuality`, `removeQuality`, `moveQuality`, `validateProfile`
- `tests/unit/ui/profile-editor.spec.ts` — 40 vitest tests (written TDD-first, all green)
- `couchpotato/ui/templates/partials/settings/profiles.html` — Alpine.js component
- `couchpotato/ui/templates/settings.html` — Profiles tab render block
- `couchpotato/ui/templates/partials/settings/scripts.html` — "Profiles" in tabOrder + tabLabels
- `couchpotato/ui/__init__.py` — `/partial/settings/profiles` FastAPI route
- `tests/e2e/profiles.spec.ts` — Playwright e2e (list, create, edit, reorder, delete+confirm)

## Tests Added

- **Unit (vitest):** 40 tests across all 6 pure functions — round-trip mapping, add/remove/reorder quality, validation edge cases (null labels, empty types), payload builder
- **E2E (Playwright):** 7 scenarios — load list, create profile, edit+Escape, reorder qualities by keyboard, delete confirm dialog, delete Escape cancel, validation error display

## Design System Conformance

- `cp.*` colour tokens throughout
- Heroicons (outline, 24×24, stroke-width=1.5) for all icons
- Modal: `role="dialog"`, `aria-modal="true"`, Escape closes, focus trap on Tab
- Delete confirm: separate `role="dialog"`, focus set to Cancel button on open
- All controls: `aria-label` or associated `<label>`
- User content via `x-text` only (never `x-html`)
- Spinner during async, toast (`aria-live="polite"`) for success/error
- No new frameworks, npm packages, CSS-in-JS, or component libraries

## What is Deferred

- Profile hide/show toggle (checkbox in ordering list from legacy UI) — profiles are always visible; `hide` flag is preserved on reorder but not surfaced in UI yet
- Drag-and-drop reorder — keyboard up/down buttons are provided instead (more accessible)
- Inline quality size editing (the "Sizes" section from legacy) — separate task

Ref: `specs/UI-MIGRATION.md` (Critical backlog item #1: quality-profile mgmt)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01EiFQkXSsY8U3hHVZi6pwoi

### CODE REVIEW

File: couchpotato/ui/templates/partials/settings/profiles.html [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+    deleteConfirmOpen: false,
+    profileToDelete: null,
+    saving: false,
+    deleting: false,
+    toastMessage: '',
+    toastType: 'success',
+    validationErrors: [],
+    selectedQualityToAdd: '',
+    formState: {},
+    _uid: Math.random().toString(36).slice(2, 8),
```
[the remark this answers is not in the document]
Comment (bassings) [reply couchpotato/ui/templates/partials/settings/profiles.html]: Fixed in 50441603 — `_uid` now uses `crypto.getRandomValues(new Uint8Array(4))` (same 8-hex-char shape) with a comment explaining why. It's only an ARIA/label id prefix, but the gate is required, so this clears alert 84. CodeQL re-running.

File: couchpotato/ui/templates/partials/settings/profiles.html [hunk 2]
```diff
@@ -0,0 +1,428 @@
+{# Quality Profiles Settings Partial #}
+{# Alpine.js component: profileEditor() — defined in partials/settings/scripts.html (classic script) #}
+{# API used: profile.list, quality.list, profile.save, profile.save_order, profile.delete #}
+{# Visibility/init mirror the logs tab: x-show + x-data on the panel, lazy init via $watch. #}
+
+<div x-show="activeTab === 'profiles'"
```
[the remark this answers is not in the document]
Comment (bassings) [reply couchpotato/ui/templates/partials/settings/profiles.html:6]: Rejecting this — it's empirically false and reverses round-10's own advice.

**Round-10 told me the opposite:** that `$watch('$root.activeTab')` is dead ($root resolves to this component's own element, so `.activeTab` is undefined) and to use `$watch('activeTab')`. I did, and **verified it in a real browser**: before opening the Profiles tab `loaded=false`; after, the watch fires and `load()` populates 18 profiles. The `profiles tab loads and shows profile list` e2e passes against this exact commit (11eb5ead) in 3.2s, and `ui-e2e-tests` is green in CI.

`$watch('activeTab', …)` resolves `activeTab` through Alpine's scope chain to the parent `settingsPanel()` (the same mechanism that makes `x-show="activeTab === 'profiles'"` on this element work). Switching back to `$root.activeTab` would re-introduce the dead watch round-10 correctly flagged. Keeping the verified-working code.

**Comment by bassings** [comment]
## Remediation summary — ready to merge

This PR went through an unusually deep agent-review cycle (11 rounds). The gate caught and fixed genuine bugs; later rounds tapered into nits and a few incorrect findings. All findings are now **addressed**, **rejected with evidence**, or **deferred per the reviewer's own "non-blocking" note**. Every fix was verified against a real server using the local Playwright loop.

### Real bugs caught & fixed (with regression tests)
- **CRITICAL** — profile reorder was non-functional: `ids[]`/`hidden[]` repeated form keys collapse under `dict(request.form())`. Now uses indexed `ids[i]` keys; e2e asserts reorder **persists across reload**.
- **CRITICAL** — profile save was broken: `types` sent as a JSON string was iterated char-by-char server-side. Now uses indexed bracket keys.
- **HIGH** — new profiles landed at `order=999`: `formToPayload` now sets order AND `saveProfile` actually sends it (the fix needed both halves; e2e asserts order < 999 via the API).
- **UX** — core (built-in) profiles' Delete button is now disabled; header save-chrome no longer leaks onto the Profiles tab (incomplete `customPanelTabs` refactor); delete dialog stays open on failure; lazy-load is now genuinely lazy (verified: `loaded=false` until the tab is opened).
- Plus: `resp.ok` checks, NaN-coalescing, label trimming, focus-trap recapture, design-system icon conformance, and accessibility (axe) green.

### Tests added: 94 vitest + 11 Playwright e2e (all green, stable serially = CI mode).

### Findings NOT actioned (with rationale)
- **"JSDoc violates a CLAUDE.md no-comment rule"** (raised 3×): no such rule exists in `CLAUDE.md`/`AGENTS.md`/`.github` — the quoted text is fabricated, and every sibling `ui/` module uses JSDoc. Per *match the surrounding code*, the JSDoc stays.
- **"$watch targets wrong scope; tab will never load"** (round 11): empirically false and reverses round 10's own advice. Verified in-browser that `$watch('activeTab')` fires and loads 18 profiles; the `profiles tab loads` e2e passes on this commit.
- **"!qualityId rejects numeric 0"**: quality identifiers are strings; 0 is not a valid input.
- **"customPanelTabs needs a data-driven registry"**: reviewer flagged as a non-blocking follow-up; deferred.

All 15 required checks (tests 3.10–3.13, ui-e2e, CodeQL, accessibility, docker, lint, dependency-review, claude-review) are green.
